### ISSUE TITLE
vuiet-artist-info tab-completion fails due to nested minibuffer

### ISSUE THREAD

**Issue by catern** [mihaiolteanu/vuiet#59]
If I run `M-x vuiet-artist-info` and hit TAB, I get this error message:
```
ivy-read: Command attempted to use minibuffer while in minibuffer
```
From looking at vuiet.el it seems that you're trying to run ivy-read while already in a read-from-minibuffer. I guess that doesn't work?

**Comment by mihaiolteanu** [comment]
I don't get that behavior.
Provide more context or relevant info.

You can also try `(vuiet--artist-from-minibuffer-if-nil '())` directly to test if you get the same result.

**Comment by mihaiolteanu** [comment]
Do you otherwise have a working configuration, meaning,
```
(vuiet-artist-info "queen")
```
works as expected?

**Comment by catern** [comment]
I have vuiet version 20210216.751 from MELPA, ivy version 20210311.1638 from MELPA.

The `ivy-read: Command attempted to use minibuffer while in minibuffer` error also happens with `(vuiet--artist-from-minibuffer-if-nil '())`

```
(vuiet-artist-info "queen")
```

shows a buffer full of information about Queen (as expected)

**Comment by catern** [comment]
BTW, you might consider using `completing-read` instead of hardcoding ivy; `completing-read` can be customized by the user to use ivy or other packages with `completing-read-function`.

**Comment by catern** [comment]
I get this behavior even while running with `emacs -q`; `M-x package-initialize` followed by `M-x vuiet-artist-info TAB` gives the error.

**Comment by mihaiolteanu** [comment]
I can't confirm it with the latest ivy (the latest vuiet I've already had).

**Comment by mihaiolteanu** [comment]
Found it.
Do you have  `enable-recursive-minibuffers` set to `t`?
https://www.gnu.org/software/emacs/manual/html_node/elisp/Recursive-Mini.html

**Comment by catern** [comment]
Makes sense; enable-recursive-minibuffers is set to the default nil for me.

If I add `(enable-recursive-minibuffers t)` to the let-binding in vuiet--artist-from-minibuffer-if-nil, it works now.

**Comment by mihaiolteanu** [comment]
I've had this same issue with the minibuffers while developing vuiet.
Then I've set the `enable-recursive-minibuffers` in my init.el and forgot about it.

> If I add (enable-recursive-minibuffers t) to the let-binding in vuiet--artist-from-minibuffer-if-nil, it works now.

If you can find a nice solution to it and make a PR, I'll consider it.

Thanks for your contribution.

**Comment by mihaiolteanu** [comment]
Fixed with d881601
